tensorrt_llm/_torch/modules/fused_moe/quantization.py

```diff
@@ -2085,6 +2085,29 @@ def load_quant_scales(self, module: torch.nn.Module, weights: Dict):
                                        non_blocking=True)
 
 
+def _get_weight_alignment(weight_alignment, scaling_vector_size, tp_size,
+                          shard_dim_size):
+
+    def lcm(a, b):
+        return abs(a * b) // math.gcd(a, b)
+
+    # The alignment should be the least common multiple of weight_alignment, scaling_vector_size,
+    # and tp_size. scaling_vector_size and tp_size must be considered
+    # to avoid fractional scaling factors.
+    alignment = lcm(weight_alignment, scaling_vector_size)
+    alignment = lcm(alignment, tp_size)
+
+    # If after the alignment, the sharding dim per shard is not a multiple of weight_alignment,
+    # we need to pad the weights to make it a multiple of weight_alignment.
+    padded_weights_dim = math.ceil(shard_dim_size / alignment) * alignment
+    per_shard = padded_weights_dim // tp_size
+    if per_shard % weight_alignment != 0:
+        alignment = weight_alignment * math.ceil(
+            per_shard / weight_alignment) * tp_size
```

Comment:
```suggestion
    def _align(dim, alignment):
        return math.ceil(dim / alignment) * alignment
    
    return _align(shard_dim_size, tp_size * weight_alignment)
```
This code can be simplified.

Reply:
Looks like entire function is the same as` math.ceil(shard_dim_size/ (weight_alignment*tp_size)) * weight_alignment*tp_size` if we know shard_dim_size is 2880, tp size is 1 2 4 8, scaling_vector_size is 32, weight_alignment is 128 or 256 because lcms are no op here.